[Docs][9.0.2] Add known issue about kibana.yml file (#222213)
elastic/kibana: docs/release-notes/known-issues.md

# Kibana known issues

::::{dropdown} Upgrading Kibana from 8.18.x to 9.0.2 fails due to a configuration conflict in the kibana.yml file

Applies to: {{stack}} 9.0.2

**Details**

> Upgrading {{kib}} from version 8.18.x to 9.0.2 fails due to a configuration conflict in the `kibana.yml` file. {{kib}} fails to boot and shows a fatal error message in the {kib} logs:

Reviewer: ```suggestion
Upgrading {{kib}} from version 8.18.x to 9.0.2 fails due to a configuration conflict if `xpack.alerting.cancelAlertsOnRuleTimeout` is set to `false` in the `kibana.yml` file. {{kib}} fails to boot and shows a fatal error message in the {kib} logs:
```

````
FATAL Error: Rule type "transform_health" cannot have both cancelAlertsOnRuleTimeout set to false and autoRecoverAlerts set to true.
````

> This failure occurs because the `xpack.alerting.cancelAlertsOnRuleTimeout` setting is set to `false`, which is incompatible with the default setting of `autoRecoverAlerts`, which is set to `true` in 9.0.2. 

Reviewer: ```suggestion
This failure occurs when the `xpack.alerting.cancelAlertsOnRuleTimeout` setting is set to `false`, which is incompatible with the default setting of `autoRecoverAlerts`, which is set to `true` in 9.0.2. 
```

> This failure occurs when the `xpack.alerting.cancelAlertsOnRuleTimeout` setting is set to `false`, which is incompatible with the default setting of `autoRecoverAlerts`, which is set to `true` in 9.0.2. 

Author: ```suggestion
This failure occurs when the `xpack.alerting.cancelAlertsOnRuleTimeout` setting is set to `false`, which is incompatible with the default configuration of an internal setting (`autoRecoverAlerts`) in 9.0.2.
```
